adafruit_trellism4.py

```diff
@@ -82,8 +83,53 @@ def __setitem__(self, index, value):
             raise IndexError("Pixel assignment outside available coordinates.")
 
         self._neopixel[offset] = value
+        if self._auto_write:
+            self.show()
+
+    def show(self):
+        """
+        Pass through to underlying NeoPixel object.
+        For use when auto_write == False
+
+        Shows the new colors on the pixels themselves if they haven't already
+        been autowritten.
+        The colors may or may not be showing after this function returns because
+        it may be done asynchronously.
+        """
         self._neopixel.show()
 
+    @property
+    def auto_write(self):
+        """
+        True if the neopixels should immediately change when set. If False,
+        `show` must be called explicitly.
+
+        This example disables auto_write, sets every pixel, calls show(), then
+        re-enables auto-write. trellis.pixels.fill() should really be used here
+        since all pixels are the same, but this is just for the sake of example.
+
+        .. code-block:: python
+
+            import adafruit_trellism4
+
+            trellis = adafruit_trellism4.TrellisM4Express()
+
+            trellis.pixels.auto_write = False
+
+            for x in range(trellis.pixels.width):
+                for y in range(trellis.pixels.height):
+                    trellis.pixels[x, y] = (0, 255, 0)
+
+            trellis.pixels.show() # must call show() when auto_write == False
+
+            trellis.pixels.auto_write = True
+        """
+        return self._auto_write
```

Comment:
Why not just have this be a pass thrus to the underlying `neopixel` object?
```python
return self._neopixel.auto_write
```

Reply:
Because it would break `_NeoPixelArray.fill()`, as noted above I was attempting to make the change with minimal modification to how `_NeoPixelArray` already worked.